components/suggest/src/rs.rs

```diff
@@ -132,109 +126,81 @@ pub(crate) struct DownloadedAmpSuggestion {
     pub iab_category: String,
     pub click_url: String,
     pub impression_url: String,
+    #[serde(rename = "icon")]
+    pub icon_id: String,
 }
 
-/// Provider Types
-#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
-#[repr(u8)]
-pub enum SuggestionProvider {
-    Amp = 1,
-    Wikipedia = 2,
-}
-
-impl FromSql for SuggestionProvider {
-    fn column_result(value: ValueRef<'_>) -> FromSqlResult<Self> {
-        let v = value.as_i64()?;
-        u8::try_from(v)
-            .ok()
-            .and_then(SuggestionProvider::from_u8)
-            .ok_or_else(|| FromSqlError::OutOfRange(v))
-    }
-}
-
-impl SuggestionProvider {
-    #[inline]
-    pub fn from_u8(v: u8) -> Option<Self> {
-        match v {
-            1 => Some(SuggestionProvider::Amp),
-            2 => Some(SuggestionProvider::Wikipedia),
-            _ => None,
-        }
-    }
-}
-
-impl ToSql for SuggestionProvider {
-    fn to_sql(&self) -> RusqliteResult<ToSqlOutput<'_>> {
-        Ok(ToSqlOutput::from(*self as u8))
-    }
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq)]
+pub(crate) struct DownloadedWikipediaSuggestion {
+    #[serde(flatten)]
+    pub common_details: DownloadedSuggestionCommonDetails,
+    #[serde(rename = "icon")]
+    pub icon_id: String,
 }
 
-/// A suggestion to ingest from a downloaded Remote Settings attachment.
+/// A suggestion to ingest from an AMP-Wikipedia attachment downloaded from
+/// Remote Settings.
 #[derive(Clone, Debug, Eq, PartialEq)]
-pub(crate) enum DownloadedSuggestion {
+pub(crate) enum DownloadedAmpWikipediaSuggestion {
```

Comment:
Are we looking to add more suggestion types (as enum variants) here in the future? If so, having `Amp` and `Wikipedia` in the type identifier seems a bit inflexible to me. `DownloadedSuggestion` sounds perfect to me. If we will introduce new enums instead, plz disregard my comment :)

Reply:
I'm glad you noticed this! 😄 I was thinking of introducing new structs for the new types, which is why I renamed it, and the `SuggestDao` method, to be specific to AMP-Wikipedia.

IIUC, AMP-Wikipedia attachments have the same Remote Settings record type (`data`), but the newer ones have different record types (`amo-suggestions` and `pocket-suggestions`). We could definitely have variants for those in `DownloadedSuggestion`, but I mostly wanted to avoid that for simplicity—we can avoid complicating the custom deserializer with cases for these other different types.

Reply:
Ah, I see. So we're modelling against those record types here. Makes sense, let's leave it as is then. Maybe add a quick comment here for future reference.

Reply:
Added some docs for how to add new suggestion types in https://github.com/mozilla/application-services/pull/5766/commits/839f4990dc395ad471fe9e08199599366009acd7 😁👍🏼 (/cc @tiftran)